src/containers/ImportResources/ImportResources.test.js

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { render, fireEvent } from 'react-testing-library';
+import { Provider } from 'react-redux';
+import thunk from 'redux-thunk';
+import configureStore from 'redux-mock-store';
+import * as ImportResources from './ImportResources';
+import * as API from '../../api';
+
+beforeEach(jest.resetAllMocks);
+
+describe ('ImportResources component', () => {
+  const middleware = [thunk];
+  const mockStore = configureStore(middleware);
+  const store = mockStore({ 
+    namespaces: { selected: 'default' },
+    serviceAccounts: {
+      byNamespace: ['default'],
+      errorMessage: null,
+      isFetching: false
+    } 
+  });
+
+  it('Valid data submit displays success notification ', async () => {
+    const headerKey = 'Content-Location'
+    const headers = {
+      get(headerKey) {
+        return 'fake-tekton-pipeline-run';
+      }
+     }
+    
+    jest.spyOn(API, 'createPipelineRun').mockImplementation(() => Promise.resolve(headers));
+    const { queryByText, getByTestId, getByText } = render(<Provider store={store} ><ImportResources.ImportResources /></Provider>);
+
+    const repoURLField = getByTestId('repository-url-field');
+    await fireEvent.change(repoURLField, { target: { value: 'URL' } });
```

Comment:
`fireEvent` is synchronous, no need for async/await

Reply:
if I remove all async/await's in my tests they start failing, is this unexpected behaviour?

Reply:
instead of using async/await (which is introducing a slight delay and just happens to be enough for the component to update) we should wait for the update to happen, something like this:

```js
  it('Valid data submit displays success notification ', () => {
    const headers = {
      get() {
        return 'fake-tekton-pipeline-run';
      }
    };

    jest
      .spyOn(API, 'createPipelineRun')
      .mockImplementation(() => Promise.resolve(headers));
    const { getByTestId, getByText } = render(
      <Provider store={store}>
        <ImportResources.ImportResources />
      </Provider>
    );

    const repoURLField = getByTestId('repository-url-field');
    fireEvent.change(repoURLField, { target: { value: 'URL' } });
    fireEvent.click(getByText('Import and Apply'));
    waitForElement(() =>
      getByText(/Triggered PipelineRun to apply Tekton resources/i)
    );
  });
```

verified locally that this passes consistently


I also noticed quite a few lint errors in that file, it'd be a good idea to `npm run lint` and address any that remain. `npm run lint:fix` will automatically fix any whitespace / layout related issues which usually account for the majority. I'll also take a look at why this isn't being picked up in the PR builds.

Reply:
ok thanks Alan! I think it is that none of the test files are being linted, as when I am doing npm start locally it is doing all the correct linting for me in all files but not in the test files

Reply:
I am having some trouble with this linting error:
``` 
const headerKey = 'Content-Location';
const headers = {
  get(headerKey) {
    return 'fake-tekton-pipeline-run';
  }
};
```

it is reporting that header key is defined and not used when it is used here in my code
```
  35:9   error  'headerKey' is assigned a value but never used      no-unused-vars
  39:11  error  'headerKey' is already declared in the upper scope  no-shadow
  39:11  error  'headerKey' is defined but never used               no-unused-vars
```

Reply:
`headerKey` isn't used in that snippet so you could just have:

```js
const headers = {
  get() {
    return 'fake-tekton-pipeline-run';
  }
};
```

That'll resolve the linting errors and still does the exact same thing.